src/shared.ts

```diff
@@ -645,16 +646,31 @@ export async function BlockBlueVerified(user: BlueBlockerUser, config: Config) {
 			return;
 		}
 
+		// Step 0: Check for disallowed words or emojis in usernames.
+		const disallowedWordsWithoutEmptyStrings = config.disallowedWords.filter(word => word !== '');
+		if (disallowedWordsWithoutEmptyStrings.length > 0){
+			// this makes extra sure that emojis are always detected, regardless if they are attached to a word or another string of emojis.
+			// the 'i' makes the test case insensitive, which helps users not have to worry about typing the same word multiple times with different variations.
+			const disallowedWordsAndEmojis = new RegExp(`${config.disallowedWords.map(word=>word.split('').join(' *')).join('(?= |$)|')}(?= |$)`, 'i');
+			const usernameToTest = (user.legacy.name).replace(/(?<= ) +/, '');
```

Comment:
I am having a really hard time discerning why all of this is necessary

```js
>> `${["one", "two", "three"].map(word=>word.split('').join(' *')).join('(?= |$)|')}(?= |$)`
"o *n *e(?= |$)|t *w *o(?= |$)|t *h *r *e *e(?= |$)" 
```

this actually seems like the opposite of what we'd want?

I'm also not sure why it's necessary to modify the username. it looks like you're trying to replace multiple spaces with a single space but in a very roundabout way, `replace(/ {2,}/, ' ')` would be much more effective, but I still don't know why it's necessary?

Reply:
In terms of the adding a space between characters part, I was mostly hypothesizing that some Twitter users might try to do "B A D W O R D", as a stylistic choice but I'm not sure how realistic that is?

For the editing usernames, same kinda deal. Your regex is a lot better, I honestly was kinda lost in the sauce of lookahead/behind groups lmao.